lib/src/api.dart

```diff
@@ -80,228 +79,134 @@ class AppStoreConnectApi {
   }
 
   Future<InAppPurchase> getInAppPurchase(String id) async {
-    final response = await _client.get(GetRequest(AppStoreConnectUri.v2(resource: 'inAppPurchases/$id')));
+    final response = await _client.get(GetRequest(AppStoreConnectUri.v2('inAppPurchases/$id')));
     return response.as<InAppPurchase>();
   }
 
   Future<List<InAppPurchase>> getInAppPurchases({int limit = 200}) async {
     final request = GetRequest(AppStoreConnectUri.v1('apps/$_appId/inAppPurchasesV2'))..limit(limit);
-    print('uri: ${request.toUri()}');
     final response = await _client.get(request);
 
     return response.asList<InAppPurchase>()..sort((a, b) => a.productId.compareTo(b.productId));
   }
 
-  Future<bool> postInAppPurchase(String productId, String name, String reviewNote, InAppPurchaseType type) async {
-    final body = {
-      'data': {
-        'type': 'inAppPurchases',
-        'attributes': {
-          'familySharable': false,
-          'inAppPurchaseType': type.toString(),
-          'name': name,
-          'productId': productId,
-          "reviewNote": reviewNote,
-        },
-        'relationships': {
-          'app': {
-            'data': {
-              'type': 'apps',
-              'id': _appId,
-            }
-          }
-        }
-      }
-    };
-
-    await _client.post(AppStoreConnectUri.v2(resource: 'inAppPurchases'), body);
+  Future<bool> postInAppPurchase(InAppPurchaseAttributes attributes) async {
+    await _client.postModel(
+      AppStoreConnectUri.v2(),
+      'inAppPurchases',
+      attributes: attributes,
+      relationships: {
+        'app': SingleModelRelationship(type: 'apps', id: _appId),
+      },
+    );
```

Comment:
[Anmerkung] Der Post würde dir ja in der Response schon das erstellte InAppPruchase Objekt zurückgeben. Wenn du das hier zurückgibst, musst du im Mate Command nicht wieder alle InAppPurchases laden um anhand der Product ID die IAP ID aufzulösen.

Reply:
Gute Idee. done.